qcodes/dataset/database_copy_paste.py

```diff
@@ -0,0 +1,397 @@
+from typing import Union
+
+import numpy as np
+
+from qcodes.dataset.data_set import DataSet
+from qcodes.dataset.experiment_container import load_or_create_experiment
+from qcodes.dataset.sqlite_base import (atomic,
+                                        connect,
+                                        format_table_name,
+                                        get_last_experiment,
+                                        insert_column,
+                                        SomeConnection)
+
+
+def sql_placeholder_string(n: int) -> str:
+    """
+    Return an SQL value placeholder string of length n.
+    Example: sql_placeholder_string(5) returns '(?,?,?,?,?)'
+    """
+    return '(' + ','.join('?'*n) + ')'
+
+
+def copy_runs_into_db(source_db_path: str,
+                      target_db_path: str, *run_ids) -> None:
+    """
+    Copy a selection of runs into another DB file. All runs must come from the
+    same experiment. They will be added to an experiment with the same name
+    and sample_name in the target db. If such an experiment does not exist,
+    it will be created.
+
+    Args:
+        source_db_path: Path to the source DB file
+        target_db_path: Path to the target DB file. The target DB file will be
+          created if it does not exist.
+        run_ids: The run_ids of the runs to copy into the target DB file
+    """
+
+    # Validate that all runs are from the same experiment
+
+    sql_placeholders = sql_placeholder_string(len(run_ids))
+    exp_id_query = f"""
+                    SELECT exp_id
+                    FROM runs
+                    WHERE run_id IN {sql_placeholders}
+                    """
+    source_conn = connect(source_db_path)
+    cursor = source_conn.cursor()
+    cursor.execute(exp_id_query, run_ids)
+    rows = cursor.fetchall()
+    source_exp_ids = np.unique([exp_id for row in rows for exp_id in row])
+    if len(source_exp_ids) != 1:
+        source_conn.close()
+        raise ValueError('Did not receive runs from a single experiment. '
+                         f'Got runs from experiments {source_exp_ids}')
```

Comment:
if a user provides a non-existing run_id, this exception pops up also instead of nicely saying "run_id blabla does not exist, hence cannot copy it". i suggest to improve it (for example via call to `DataSet` ;) )

Reply:
That is not quite correct, that only happens if the user **only** provides a non-existing `run_id`, since then `len(source_exp_ids) == 0`. If an invalid `run_id` sneaks in among a bunch of valid ones, another error is thrown. But you are of course right that the `run_ids` should be validated up front.